dragonchain/lib/dto/bnb.py

```diff
@@ -0,0 +1,372 @@
+BLOCK_THRESHOLD = 3  # The number of blocks that can pass by before trying to send another transaction
+SEND_FEE = 37500  # transfer fee fixed at 0.000375 BNB : https://docs.binance.org/trading-spec.html#current-fees-table-on-mainnet
+
+_log = logger.get_logger()
+
+
+def new_from_user_input(user_input: Dict[str, Any]) -> "BinanceNetwork":
+    """Create a new BinanceNetwork model from user input
+    Args:
+        user_input: User dictionary input (assumed already passing create_binance_interchain_schema)
+    Returns:
+        Instantiated BinanceNetwork client
+    Raises:
+        exceptions.BadRequest: With bad input
+    """
+    dto_version = user_input.get("version")
+    if dto_version == "1":
+        if not user_input.get("private_key"):
+            # We need to create a private key if not provided
+            user_input["private_key"] = base64.b64encode(secp256k1.PrivateKey().private_key).decode("ascii")
+        else:
+            try:
+                # Check if user provided key is hex and convert if necessary
+                if len(user_input["private_key"]) == 66:  # private keys in hex are 66 chars with leading 0x
+                    user_input["private_key"] = user_input["private_key"][2:]  # Trim the 0x
+                if len(user_input["private_key"]) == 64:  # private keys in hex are 64 chars
+                    user_input["private_key"] = base64.b64encode(bytes.fromhex(user_input["private_key"])).decode("ascii")
+                else:  # assume key is a mnemonic string
+                    seed = mnemonic.Mnemonic.to_seed(mnemonic)
+                    parent_wallet = network.keys.bip32_seed(seed)
+                    child_wallet = parent_wallet.subkey_for_path("44'/714'/0'/0/0")
+                    # convert secret exponent (private key) int to hex
+                    key_hex = format(child_wallet.secret_exponent(), "x")
+                    user_input["private_key"] = base64.b64encode(bytes.fromhex(key_hex)).decode("ascii")
+            except Exception:
+                # If there's an error here, it's a bad key. Sets it to a bad key, will be caught later when making the client
+                user_input["private_key"] = "BAD_KEY_WAS_FOUND"
+
+        # check for user-provided node address
+        if not user_input.get("node_ip"):
+            # default to Dragonchain managed Binance node if not provided
+            user_input["node_ip"] = NODE_IP
+            user_input["rpc_port"] = TESTNET_RPC_PORT if user_input.get("testnet") else MAINNET_RPC_PORT
+            user_input["api_port"] = TESTNET_API_PORT if user_input.get("testnet") else MAINNET_API_PORT
+        else:
+            # accept port #s from user, then fix formatting behind the scenes for them to work
+            user_input["rpc_port"] = user_input["rpc_port"]
+            user_input["api_port"] = user_input["api_port"]
+        # check if user specified that node is a testnet
+        if user_input.get("testnet") is None:
```

Comment:
Fixed -- `testnet` value checked before user-input node address is checked.  Looks like it could fall over if user specifies a node address but not ports, however.  Suggestions for conditionally handling this?

Reply:
Fixed.